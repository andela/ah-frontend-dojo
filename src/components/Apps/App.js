import React, { Component } from 'react';
// import Routes from 'pages/Route';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Login from 'components/Login';
import LandingPage from 'pages/Landing';
import PageNotFound from 'pages/Error';

import 'assets/MainStyle.scss';

class App extends Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" render={props => <LandingPage {...props} />} />
          <Route path="/login" component={Login} />
          <Route component={PageNotFound} />
        </Switch>
      </Router>
    );
  }
}
export default App;
