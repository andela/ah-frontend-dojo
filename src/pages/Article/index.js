/* eslint-disable no-shadow */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable jsx-a11y/img-redundant-alt */
// react libraries
import React, { Component } from 'react';

// styles
import './ArticlePage.scss';

// components
import Navbar from 'components/NavBar';
import Search from 'components/Search';
import Footer from 'components/Footer';
import { connect } from 'react-redux';
import ArticleCardLarge from 'components/ArticleCards';
import { getOneArticle } from 'store/actions/articleActions';
import { defaultImgURL } from 'utils';

export class ArticlePage extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/prop-types
    const { getArticle, match: { params: { slug } } } = this.props;
    getArticle(slug);
  }

  fetchArticle = (e) => {
    e.preventDefault();
    // const { title, description, body } = this.state;
  };

  render() {
    // eslint-disable-next-line react/prop-types
    const { article } = this.props;
    return (
      <div className="canvas">
        <div className="landing">
          <Navbar />
        </div>
        <div className="breadcrumbs" align="left">
          <ul>
            <li>
              <img src={`${defaultImgURL}/images/breadcrumbs.png`} alt="breadcrumbs" />
              /
              <a href="/articles"><b>Articles</b></a>
              /Article/
            </li>
          </ul>
        </div>
        <Search />
        <div align="center" className="articlesCanvas">
          <div className="articleContainer">
            <ArticleCardLarge {...article} />
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { article } = state.articleReducer;
  return { article };
};

const mapDispatchToProps = slug => dispatch => ({
  getArticle(slug) {
    dispatch(getOneArticle(slug));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ArticlePage);
