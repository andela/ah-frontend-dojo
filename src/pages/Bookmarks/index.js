// react
import React, { Component } from 'react';

// 3rd party libraries
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// styles
import './Bookmarks.scss';

// actions
import { toggleBookmark, setInitialBookmarkStatus } from 'store/actions/bookmarkActions';


export class Bookmarks extends Component {
  componentDidMount() {
    const { getInitialBookmarkValue } = this.props;
    getInitialBookmarkValue('building-world-class-features');
  }

  handleClick = (e) => {
    e.preventDefault();
    const { dispatchToggleBookmarks } = this.props;
    dispatchToggleBookmarks('building-world-class-features');
  }

  render() {
    const { isBookmarked } = this.props;

    return (
      <div className="bookmarks">
        <div className="bookmarks__header">
          <p className="bookmarks__header__title">Bookmarks</p>
        </div>
        {
          isBookmarked
            ? (
              <div
                onClick={this.handleClick}
                tabIndex="0"
                role="button"
                onKeyPress={this.handleKeyPress}
                className="bookmarks__isBookmarked"
              />
            )
            : (
              <div
                onClick={this.handleClick}
                tabIndex="0"
                role="button"
                onKeyPress={this.handleKeyPress}
                className="bookmarks__notBookmarked"
              />
            )
        }
      </div>
    );
  }
}

// prop type validation
Bookmarks.propTypes = {
  dispatchToggleBookmarks: PropTypes.func,
  getInitialBookmarkValue: PropTypes.func,
  isBookmarked: PropTypes.bool,
};

// default props
Bookmarks.defaultProps = {
  dispatchToggleBookmarks: () => { },
  getInitialBookmarkValue: () => { },
  isBookmarked: false,
};

export const mapStateToProps = ({
  dispatchToggleBookmarks,
  getInitialBookmarkValue,
  bookmarkReducer: {
    isBookmarked,
  },
}) => ({
  getInitialBookmarkValue,
  dispatchToggleBookmarks,
  isBookmarked,
});

export const mapDispatchToProps = dispatch => ({
  dispatchToggleBookmarks: slug => dispatch(toggleBookmark(slug)),
  getInitialBookmarkValue: slug => dispatch(setInitialBookmarkStatus(slug)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Bookmarks);
