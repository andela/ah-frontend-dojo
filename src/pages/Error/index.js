import React from 'react';
// styles
import './ErrorPageStyle.scss';

const PageNotFound = () => (
  <div className="container">
    <div className="copy-container error-space">
      <p className="error">
        404
      </p>
      <br />
      <p className="error1">
        Page not found.
      </p>
      <p className="error2">
        <a href="https://ah-frontend-dojo.herokuapp.com/">Go back</a>
      </p>
    </div>
  </div>

);
export default PageNotFound;
