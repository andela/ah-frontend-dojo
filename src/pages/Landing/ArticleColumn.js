// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// components
import ArticlePreview from 'components/ArticlePreview';


const ArticleColumn = (props) => {
  const { columnTitle, articles } = props;
  return (
    <div className="article-grid__col">
      <div className="article-grid__col__header">
        <p className="article-grid__col__header__title">{columnTitle}</p>
      </div>
      {
        articles.map(article => <ArticlePreview key={article.id} article={article} />)
      }
    </div>
  );
};

ArticleColumn.propTypes = {
  columnTitle: PropTypes.string.isRequired,
  articles: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    slug: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publish_status: PropTypes.bool.isRequired,
    createdAt: PropTypes.string.isRequired,
    updatedAt: PropTypes.string.isRequired,
    delete_status: PropTypes.bool.isRequired,
    tagList: PropTypes.arrayOf(
      PropTypes.string,
    ),
    time_to_read: PropTypes.number.isRequired,
    read_stats: PropTypes.shape({
      views: PropTypes.number.isRequired,
      reads: PropTypes.number.isRequired,
    }),
    likeCount: PropTypes.arrayOf(PropTypes.shape({
      likes: PropTypes.number.isRequired,
      dislikes: PropTypes.number.isRequired,
    })),
  })),
};

ArticleColumn.defaultProps = {
  articles: [],
};

export default ArticleColumn;
