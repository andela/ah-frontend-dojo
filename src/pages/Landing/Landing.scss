@import url("https://fonts.googleapis.com/css?family=Open+Sans:600&display=swap");
$primary: #2D6686;
$primary-ultra-light: #DAE8F2;
$secondary: #E91E63;
$shadow: #D3D6DB;
$white: #FFFFFF;

.banner-image {
  margin-top: 3px;
  margin-bottom: 0.3rem;
  background: url("../../assets/images/perfecto-capucine-779224-unsplash.jpg");
  width: 100%;
  height: 27rem;
  overflow: hidden;
  background-size: cover;
  background-position: center;
  box-shadow: 0.1rem 0.1rem 0.7rem $shadow;
}

.welcome-pitch {
  float: left;
  margin: auto;
  margin-bottom: 2rem;
  width: 100%;
  color: $primary;

  &__left {
    float: left;
    width: 50%;
    padding-top: 0.5rem;
    padding-right: 1rem;
    border-right: 0.05rem solid $secondary;
    text-align: right;
    font-size: 1.7rem;
  }

  &__right {
    float: left;
    width: 50%;
    padding-top: 0.5rem;
    padding-left: 2rem;
    text-align: left;
    font-size: 1.2rem;
  }
}

.article-grid {
  display: flex;
  margin: auto;
  margin-top: 1rem;
  width: 90%;

  &__col {
    margin-left: 2rem;
    float: left;
    width: 30%;

    &__header {
      height: 2.5rem;
      text-align: center;
      border-top: 0.3rem solid $primary;

      &__title {
        padding-top: 0.3rem;
        font-size: 1.5rem;
        color: $secondary;
      }
    }
  }
}

.article-preview {
  margin: auto;
  margin-top: 0.5rem;
  margin-bottom: 1rem;
  width: 90%;
  box-shadow: 0.1rem 0.1rem 0.7rem $shadow;

  &__title {
    padding: 0.5rem;
    font-family: "Open Sans", sans-serif;
    color: $primary;
  }

  &__sample {
    padding: 0.5rem;
    height: 3rem;
    overflow: hidden;
  }

  &__image {
    width: 100%;
    padding: 0.8rem;
    height: 7rem;
    overflow: hidden;
    background-size: cover;
    background-position: center;
    background-image: url("../../assets/images/perfecto-capucine-779224-unsplash.jpg");
  }

  &__read-time {
    padding: 0.5rem;
  }

  &__footer {
    height: 3rem;
    padding: 0.8rem;
    background-color: $primary;
    color: $white;

    &__author {
      float: left;
      width: 40%;
    }

    &__date {
      float: right;
    }
  }
}

@media screen and (max-width: 600px) {
  .article-grid {
    width: 100%;

    &__col {
      width: 90%;
    }
  }

  &__col {
    margin-left: 2rem;
    /* float: left; */
    width: 80%;
  }
}

@media screen and (max-width: 1066px) {
  .article-grid__col {
    margin-left: 0rem;
  }
}
