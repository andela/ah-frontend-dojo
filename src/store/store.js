import thunk from 'redux-thunk';
import { createStore, applyMiddleware, compose } from 'redux';
import reducer from './rootReducer';

const enhancers = compose(
  applyMiddleware(thunk),
  window.devToolsExtension && window.devToolsExtension(),
);
const store = createStore(reducer, enhancers);

export default store;
