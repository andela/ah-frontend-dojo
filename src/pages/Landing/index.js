// react libraries
import React, { Component } from 'react';

// third-party libraries
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// styles
import './Landing.scss';

// components
import ArticlePreview from 'components/ArticlePreview';
import Navbar from 'components/NavBar/NavBar';
import Footer from 'components/Footer';

// thunks (action creators)
import { getAllArticles } from 'store/actions/articleActions';


export class LandingPage extends Component {
  componentDidMount() {
    const { getArticles } = this.props;
    if (getArticles) {
      getArticles();
    }
  }

  render() {
    const { articles } = this.props;
    return (
      // Navbar
      <div>
        <div className="landing">
          <Navbar />
          <div className="banner-image" />

          {/* Welcome message */}
          <div className="welcome-pitch">
            <div className="welcome-pitch__left">
              <p>
                Welcome to
              </p>
              <h3>
                Authors Haven
              </h3>
            </div>
            <div className="welcome-pitch__right">
              <p>
                Intrigue your interests
              </p>
              <p>
                Broaden your world
              </p>
              <p>
                Tell your story
              </p>
            </div>
          </div>

          {/* Article preview grid */}
          <div className="article-grid">
            <div className="article-grid__col">
              <div className="article-grid__col__header">
                <p className="article-grid__col__header__title">Most Popular</p>
              </div>
              {
                articles.map(article => <ArticlePreview key={article.id} article={article} />)
              }
            </div>
            <div className="article-grid__col">
              <div className="article-grid__col__header">
                <p className="article-grid__col__header__title">Most Liked</p>
              </div>
              {
                articles.map(article => <ArticlePreview key={article.id} article={article} />)
              }
            </div>
            <div className="article-grid__col">
              <div className="article-grid__col__header">
                <p className="article-grid__col__header__title">Most Trending</p>
              </div>
              {
                articles.map(article => <ArticlePreview key={article.id} article={article} />)
              }
            </div>
          </div>
          <Footer />
        </div>
      </div>
    );
  }
}

// proptype validation
LandingPage.propTypes = {
  getArticles: PropTypes.func,
  articles: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    slug: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publish_status: PropTypes.bool.isRequired,
    createdAt: PropTypes.string.isRequired,
    updatedAt: PropTypes.string.isRequired,
    delete_status: PropTypes.bool.isRequired,
    tagList: PropTypes.arrayOf(
      PropTypes.string,
    ),
    time_to_read: PropTypes.number.isRequired,
    read_stats: PropTypes.shape({
      views: PropTypes.number.isRequired,
      reads: PropTypes.number.isRequired,
    }),
    likeCount: PropTypes.arrayOf(PropTypes.shape({
      likes: PropTypes.number.isRequired,
      dislikes: PropTypes.number.isRequired,
    })),
  })),
};

LandingPage.defaultProps = {
  articles: [],
  getArticles: () => { },
};

const mapStateToProps = state => state.articles;

const mapDispatchToProps = dispatch => bindActionCreators({
  getArticles: () => getAllArticles(),
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);
