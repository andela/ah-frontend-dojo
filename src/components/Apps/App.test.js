import React from 'react';

import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import store from 'store/store';

import { LandingPage } from 'pages/Landing';
import App from './App';

describe('routes using memory router', () => {
  it('should show Home component for / router (using memory router)', () => {
    const component = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>
      </Provider>,
    );
    expect(component.find(LandingPage)).toHaveLength(1);
  });
});
