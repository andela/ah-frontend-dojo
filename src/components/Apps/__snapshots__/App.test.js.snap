// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
